controllers/common/helper.go

```diff
@@ -659,13 +659,23 @@ func deleteBackupsForDatabase(
 			// Already deleting, continue to next.
 			continue
 		}
-		if err := c.Delete(ctx, &backup); err != nil {
+		if err := ForegroundDelete(ctx, &backup, c); err != nil {
```

Comment:
What if you delete the backup directly? Shouldn't it have the finalizer as well?

Reply:
Updated